liquidhaskell-boot/src/Language/Haskell/Liquid/Bare/Misc.hs

```diff
@@ -153,9 +154,13 @@ mapTyVars _ hsT lqT
   = do err <- gets errmsg
        throwError (err (F.pprint hsT) (F.pprint lqT))
 
+-- Niki believes there is a bug on the isTYPEorCONSTRAINT function in GHC-9.12.2
+-- Hardcoding "GHC.Types.Type" for now because typeKind in in the Prim module... 
```

Comment:
I don't understand the comment about `typeKind` being in some module `Prim`, but there is a function `GHC.Core.Type.typeKind`.